src/modal/Modal.ts

```diff
@@ -0,0 +1,431 @@
+                    
+					padding-left: var(--omni-modal-header-padding-left, 24px);
+					padding-top: var(--omni-modal-header-padding-top, 24px);
+					padding-right: var(--omni-modal-header-padding-right, 24px);
+					padding-bottom: var(--omni-modal-header-padding-bottom, 24px);
+					position: relative;
+				}
+
+				.header.center {
+                    justify-content: center;
+                    text-align: center;
+				}
+
+				.header.right {
+                    justify-content: right;
+					text-align: right;
+				}
+				
+				.body {
+					margin-top:0px;
+					padding: var(--omni-modal-body-padding, 24px 24px 40px 24px);
+					color: var(--omni-modal-body-font-color, var(--omni-font-color));
+					font-size: var(--omni-modal-body-font-size, var(--omni-font-size));
+                    font-family: var(--omni-modal-body-font-family, var(--omni-font-family));
+                    font-weight: var(--omni-modal-body-font-weight, var(--omni-font-weight));
+					background: var(--omni-modal-body-background, var(--omni-background-color));
+					line-height: 24px;
+                    overflow: var(--omni-modal-body-overflow, auto);
+				}
+
+                .body[no-header] {                    
+					border-top-left-radius: var(--omni-modal-no-header-body-border-top-radius, 4px);
+					border-top-right-radius: var(--omni-modal-no-header-body-border-top-radius, 4px);
+                }
+
+                .body[no-footer] {                    
+					border-bottom-left-radius: var(--omni-modal-no-footer-body-border-bottom-radius, 4px);
+					border-bottom-right-radius: var(--omni-modal-no-footer-body-border-bottom-radius, 4px);
+                }
+				
+				.footer {
+					align-self: stretch;
+					text-align: var(--omni-modal-footer-text-align, right);
+					padding: var(--omni-modal-footer-padding, 12px 12px 12px 0px);
+					color: var(--omni-modal-footer-font-color,var(--omni-font-color));
+					font-size: var(--omni-modal-footer-font-size, var(--omni-font-size));
+                    font-family: var(--omni-modal-footer-font-family, var(--omni-font-family));
+                    font-weight: var(--omni-modal-footer-font-weight, var(--omni-font-weight));
+					background: var(--omni-modal-footer-background, var(--omni-background-active-color));
+				}
+
+				@media screen and (min-width: 767px) {
+					
+					.header {
+						border-radius: var(--omni-modal-header-border-radius, 4px 4px 0px 0px);
+					}
+				}
+
+				@media screen and (max-width: 767px) {
+					
+					.container:not([no-fullscreen]) {
+						height: 100%;
+					}
+                    
+                    .body:not([no-fullscreen]) {
+                        height: inherit;
+                    }
+
+				}
+      `
+        ];
+    }
+
+    override render(): TemplateResult {
+        return html`
+        <dialog part="dialog" class="modal" role="dialog" aria-modal="true"
```

Comment:
Check indentation for template result functions, looks like tabs & spaces are mixed.